### PR TITLE
Add EdDSA support for signing JWTs

### PR DESCRIPTION
**Pull request by LucDeCaf**
This adds support for the EdDSA signing algorithm in `core_service` as requested in [this issue](https://github.com/powersync-ja/powersync-service/issues/72).

I am not super familiar with the `core_service` architecture so feedback on what needs to be added/improved would be greatly appreciated.

### CODE REVIEW

**Review by rkistner — COMMENTED** [review]
To also support EdDSA with a JWKS URI, could you also modify this bit to accept these keys?

https://github.com/powersync-ja/powersync-service/blob/a47bda4bf84c508d38be4a248e98991a1af7a334/packages/service-core/src/auth/RemoteJWKSCollector.ts#L76

That check is different from the ones in KeySpec, since we specifically do not want to allow the HS algorithms (symmetric keys) with JWKS URIs.

File: packages/service-core/test/src/auth.test.ts [hunk 1]
```diff
@@ -331,12 +342,34 @@ describe('JWT Auth', () => {
...
+      .setAudience('tests')
+      .setExpirationTime('5m')
+      .sign(signKey);
+
+    const verified = (await store.verifyJwt(signedJwt, {
+      defaultAudiences: ['tests'],
+      maxAge: '6m'
+    })) as JwtPayload & { claim: string };
+
+    expect(verified.claim).toEqual('test-claim');
```
[the remark this answers is not in the document]
Comment (LucDeCaf) [reply packages/service-core/test/src/auth.test.ts:373]: Whoops 😅

File: packages/service-core/test/src/auth.test.ts [hunk 2]
```diff
@@ -331,12 +342,34 @@ describe('JWT Auth', () => {
...
-    expect(key.kid).toEqual(publicKey.kid!);
+    expect(key.kid).toEqual(publicKeyRSA.kid!);
+  });
+
+  test('signing with EdDSA', async () => {
+    const keys = await StaticKeyCollector.importKeys([publicKeyEdDSA]);
+    const store = new KeyStore(keys);
+    const signKey = (await jose.importJWK(publicKeyEdDSA)) as jose.KeyLike;
+
+    const signedJwt = await new jose.SignJWT({ claim: 'test-claim' })
```
Comment (rkistner) [inline packages/service-core/test/src/auth.test.ts:359]: The test is failing here - it currently attempts to sign using the _public_ key, while it should sign using the _private_ key.

Comment (rkistner) [reply packages/service-core/test/src/auth.test.ts]: Thanks! Looks good now. I'll test properly next week, then we can merge.

**Review by rkistner — APPROVED** [review]
Thanks, looks great!

**Pull request merged**